src/models/GitIndex.js

```diff
@@ -198,31 +249,44 @@ export class GitIndex {
     const writer = new BufferCursor(header)
     writer.write('DIRC', 4, 'utf8')
     writer.writeUInt32BE(2)
-    writer.writeUInt32BE(this.entries.length)
-    const body = Buffer.concat(
-      this.entries.map(entry => {
-        const bpath = Buffer.from(entry.path)
-        // the fixed length + the filename + at least one null char => align by 8
-        const length = Math.ceil((62 + bpath.length + 1) / 8) * 8
-        const written = Buffer.alloc(length)
-        const writer = new BufferCursor(written)
-        const stat = normalizeStats(entry)
-        writer.writeUInt32BE(stat.ctimeSeconds)
-        writer.writeUInt32BE(stat.ctimeNanoseconds)
-        writer.writeUInt32BE(stat.mtimeSeconds)
-        writer.writeUInt32BE(stat.mtimeNanoseconds)
-        writer.writeUInt32BE(stat.dev)
-        writer.writeUInt32BE(stat.ino)
-        writer.writeUInt32BE(stat.mode)
-        writer.writeUInt32BE(stat.uid)
-        writer.writeUInt32BE(stat.gid)
-        writer.writeUInt32BE(stat.size)
-        writer.write(entry.oid, 20, 'hex')
-        writer.writeUInt16BE(renderCacheEntryFlags(entry))
-        writer.write(entry.path, bpath.length, 'utf8')
-        return written
-      })
-    )
+    writer.writeUInt32BE(this.entriesFlat.length)
+    const entryToBuffer = async entry => {
```

Comment:
This function seems to be pure, can you move it as a method to the class?

Reply:
done